pkg/operator/controller/ingress/deployment.go

```diff
@@ -623,6 +648,10 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		env = append(env, corev1.EnvVar{Name: RouterDisableHTTP2EnvName, Value: "true"})
 	}
 
+	if enabled, value := HardStopAfterIsEnabled(ci, ingressConfig); enabled && len(value) > 0 {
```

Comment:
`HardStopAfterIsEnabled` already returns `false` if `len(value) == 0`.
```suggestion
	if enabled, value := HardStopAfterIsEnabled(ci, ingressConfig); enabled {
```

Reply:
This was belt-and-braces in lieu of any future change. Emitting `hard-stop-after` (i.e., no value) would mean haproxy fails to parse the config.  I can drop it, just wary of crash looping haproxy.

Reply:
The unit test can verify that this code does the right thing if the annotation is set but has an empty value.

Reply:
Dropped.